File: tests/time-taken.py
import random
from datetime import datetime

def two_sum(nums, target):
    ans = [0, 0]
    dict_map = {}

    for index, val in enumerate(nums):
        compliment = target - val
        if compliment in dict_map:
            ans[0] = dict_map[compliment]
            ans[1] = index
            return ans
        dict_map[val] = index

    return None

def generate_random(arr_size, limit):
    return [random.randint(0, limit + 1) for _ in range(arr_size)]

def two_sum_time(times, arr_size):
    for _ in range(times):
        random_arr = generate_random(arr_size, _)
        target = random.randint(0, _ + 1)
        two_sum(random_arr, target)

if __name__ == "__main__":
    print(two_sum([2, 7, 11, 15], 9))

    start_time = datetime.now()
    two_sum_time(10000, 10000)
    end_time = datetime.now()

    total = end_time - start_time

    print(type(total))
    print(total)

File: tests/TimeTaken.java
import java.util.HashMap;
import java.util.Map;
import java.util.Random;

public class TimeTaken {

    public static int[] twoSum(int[] nums, int target) {
        int[] ans = new int[2];
        Map<Integer, Integer> dictMap = new HashMap<>();

        for (int index = 0; index < nums.length; index++) {
            int val = nums[index];
            int compliment = target - val;
            if (dictMap.containsKey(compliment)) {
                ans[0] = dictMap.get(compliment);
                ans[1] = index;
                return ans;
            }
            dictMap.put(val, index);
        }

        return null;
    }

    public static int[] generateRandom(int arrSize, int limit) {
        int[] arr = new int[arrSize];
        Random random = new Random();

        for (int i = 0; i < arrSize; i++) {
            arr[i] = random.nextInt(limit + 1);
        }

        return arr;
    }

    public static void twoSumTime(int times, int arrSize) {
        for (int i = 0; i < times; i++) {
            int[] randomArr = generateRandom(arrSize, i);
            int target = new Random().nextInt(i + 1);
            twoSum(randomArr, target);
        }
    }

    public static void main(String[] args) {
        int[] result = twoSum(new int[]{2, 7, 11, 15}, 9);
        System.out.println(java.util.Arrays.toString(result));

        long startTime = System.currentTimeMillis();
        twoSumTime(100000, 100000);
        long endTime = System.currentTimeMillis();

        long total = endTime - startTime;

        System.out.println(total);
    }
}
